### PR TITLE
[rv_dm] Move debug memory definitions into rv_dm.hjson

### PR DESCRIPTION
**Pull request by msfschaffner**
Moving the register definitions for the debug memory into the `rv_dm.hjson` file obviates the need for the separate `rv_dm_debug_mem.hjson` file for DV, and allows us to conveniently access RV_DM CSRs via the chip RAL in DV sequences at the top-level. This also removes an aliasing issue that existed in top-level sims between the window exposed in `rv_dm.hjson` and the separate RAL model that was defined by `rv_dm_debug_mem.hjson`.

This is in preparation for the `rv_dm_lc_disabled` test coming in #15428.

Signed-off-by: Michael Schaffner <msf@google.com>

### CODE REVIEW

File: hw/ip/rv_dm/data/rv_dm.hjson
Comment (sriyerg) [inline hw/ip/rv_dm/data/rv_dm.hjson:119]: We could address this comment here by parameterizing the defs below with `NrHarts` value, right?

File: hw/ip/rv_dm/data/rv_dm.hjson
Comment (sriyerg) [inline hw/ip/rv_dm/data/rv_dm.hjson:108]: I suppose this also needs to be updated?

**Review by sriyerg — COMMENTED** [review]
Thanks @msfschaffner for absorbing this into the flow.

File: hw/ip/rv_dm/data/rv_dm.hjson
Comment (msfschaffner) [reply hw/ip/rv_dm/data/rv_dm.hjson:108]: This part is actually up to date - we do not instantiated the reg_top in RTL since the registers are all implemented inside the PULP module.

File: hw/ip/rv_dm/data/rv_dm.hjson
Comment (msfschaffner) [reply hw/ip/rv_dm/data/rv_dm.hjson:119]: Good point - let me try that!

File: hw/ip/rv_dm/data/rv_dm.hjson
Comment (sriyerg) [reply hw/ip/rv_dm/data/rv_dm.hjson:108]: It will be nice to not have to check in unused code! :)

File: hw/ip/rv_dm/data/rv_dm.hjson
Comment (msfschaffner) [reply hw/ip/rv_dm/data/rv_dm.hjson]: Ok I parameterized the registers that are multiregs inside the RV_DM. This only affected `HALTED` and `RESUMING`.

File: hw/ip/rv_dm/data/rv_dm.hjson
Comment (tjaychen) [inline hw/ip/rv_dm/data/rv_dm.hjson:121]: does `NrHarts` long term need to be templated like in `rv_plic`?

**Comment by tjaychen** [comment]
looks good to me! but @sriyerg probably should approve.

File: hw/ip/rv_dm/data/rv_dm.hjson
Comment (msfschaffner) [reply hw/ip/rv_dm/data/rv_dm.hjson:121]: you mean to feed it in from the topgen configuration? probably yes - although it would be nice if we found a way to pass such top-level params into the hjson without having to make the entire IP a generated IP (e.g. if only the hjson needs modifying, but the RTL does not).

File: hw/ip/rv_dm/data/rv_dm.hjson
Comment (msfschaffner) [reply hw/ip/rv_dm/data/rv_dm.hjson:108]: yeah we have to find a way to not generate that `reg_top` - I believe CI fails right now if we don't check it in. there are other such dummy `reg_top`s in the tree, e.g. for `sram_ctrl`...

File: hw/ip/rv_dm/data/rv_dm.hjson
Comment (tjaychen) [reply hw/ip/rv_dm/data/rv_dm.hjson:121]: yeah, but this kind of feeds back to the age old parmaeterized reg problem that we have.
I think we still don't have a good solution for that outside of templating. These parameters would technically show up in `rv_dm_reg_pkg.sv` or something right?

Comment (msfschaffner) [reply hw/ip/rv_dm/data/rv_dm.hjson:121]: yeah that's right...

Comment (tjaychen) [reply hw/ip/rv_dm/data/rv_dm.hjson:121]: problem for another day :)

File: hw/ip/rv_dm/data/rv_dm.hjson
Comment (vogelpi) [inline hw/ip/rv_dm/data/rv_dm.hjson:245]: What is currently missing from the debug memory map is the `HaltAddress`, `ResumeAddress` and `ExceptionAddress` following at 0x800 - 0x808. But I guess you're aware of that. Would it make sense to mention that in the hjson somewhere?

**Review by vogelpi — APPROVED** [review]
Thanks @msfschaffner , I've reviewed the hjson and RTL changes and this looks good. Just one small question .

File: hw/ip/rv_dm/data/rv_dm.hjson
Comment (msfschaffner) [reply hw/ip/rv_dm/data/rv_dm.hjson:245]: Good catch @vogelpi - let's add these in a separate PR so that we can make progress with tests that depend on this change.

**Comment by msfschaffner** [comment]
Thanks for the reviews! @sriyerg ok if we merge this?

File: hw/ip/rv_dm/data/rv_dm.hjson
Comment (msfschaffner) [reply hw/ip/rv_dm/data/rv_dm.hjson:245]: Actually, to clarify: these addresses are the first words inside the debug ROM, an they are thus already exposed via the window here. The debug ROM window starts at byte offset `0x400 + 256 x 4 = 0x800` and exposes a region of another `0x800` bytes which is more than enough to cover the `19 x 8 = 0x98` bytes currently allocated inside the debug ROM:
https://github.com/lowRISC/opentitan/blob/8b1c3bd1c8bf2d8194b4e966fc7de77ad55ee5fe/hw/vendor/pulp_riscv_dbg/debug_rom/debug_rom.sv#L24-L47

But I can add a comment mentioning this in a follow up (don't want to re-trigger CI here ;)).

**Comment by msfschaffner** [comment]
I will move ahead and merge it for now to unblock some follow-up work.
Please let me know if something should be amended.

**Pull request merged**
